src/Adapters/Etw.Net451/Implemenetation/TraceEventExtensions.cs

```diff
@@ -0,0 +1,118 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="TraceEventExtensions.cs" company="Microsoft">
+//     Copyright (c) Microsoft Corporation. All rights reserved.
+// </copyright>
+//-----------------------------------------------------------------------
+
+namespace Microsoft.ApplicationInsights.EtwCollector.Implemenetation
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Diagnostics;
+    using System.Globalization;
+    using DataContracts;
+    using Diagnostics.Tracing;
+
+    internal static class TraceEventExtensions
+    {
+        private static Lazy<Random> random = new Lazy<Random>();
+
+        private static SeverityLevel[] traceEventLevelToSeverityLevel = new SeverityLevel[]
+        {
+            SeverityLevel.Critical,     // TraceEventLevel.Always == 0
+            SeverityLevel.Critical,     // TraceEventLevel.Critical == 1
+            SeverityLevel.Error,        // TraceEventLevel.Error == 2
+            SeverityLevel.Warning,      // TraceEventLevel.Warning == 3
+            SeverityLevel.Information,  // TraceEventLevel.Informational == 4
+            SeverityLevel.Verbose       // TraceEventLevel.Verbose == 5
+        };
+
+        public static void Track(this TraceEvent traceEvent, TelemetryClient client)
+        {
+            Debug.Assert(client != null, "Should always receive a valid client");
+
+            string formattedMessage = traceEvent.FormattedMessage;
+            TraceTelemetry telemetry = new TraceTelemetry(formattedMessage, traceEventLevelToSeverityLevel[(int)traceEvent.Level]);
+
+            telemetry.AddProperty("EventId", traceEvent.ID.ToString());
+            telemetry.AddProperty(nameof(traceEvent.EventName), traceEvent.EventName);
+            if (traceEvent.ActivityID != default(Guid))
+            {
+                telemetry.AddProperty(nameof(traceEvent.ActivityID), traceEvent.ActivityID.ToString());
```

Comment:
@SergeyKanzhelev good question that we should probably take offline. This ties to the correlation work that Liudmila is doing as well. I am more-or-less familiar with how ETW Activity IDs work but I am not sure how they tie to AI Operation IDs.